### PR TITLE
New Source dreambigtl (Dream Big Translations)

### PR DESCRIPTION
**Pull request by r1di**

### CODE REVIEW

File: src/plugins/english/dreambigtl.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  async popularNovels(
+    pageNo: number,
+    { showLatestNovels }: Plugin.PopularNovelsOptions,
+  ): Promise<Plugin.NovelItem[]> {
+    const delay = (ms: number) =>
+      new Promise(resolve => setTimeout(resolve, ms));
+
+    try {
+      const url = `${this.site}p/disclaimer.html`;
+      console.log('Fetching URL:', url);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:21]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+      await delay(2000);
+      const response = await fetchApi(url);
+      const body = await response.text();
+      console.log('Response body length:', body.length);
+
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:27]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+      const loadedCheerio = parseHTML(body);
+      const novels: Plugin.NovelItem[] = [];
+      const categories = ['New Novels', 'Ongoing Novels', 'Completed Novels'];
+
+      console.log(
+        'Menu items found:',
+        loadedCheerio('#webify-pro-main-nav-menu > li').length,
+      );
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:35]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+      loadedCheerio('#webify-pro-main-nav-menu > li.has-sub').each(
+        (_, categoryElem) => {
+          const categoryName = loadedCheerio(categoryElem)
+            .children('a')
+            .first()
+            .text()
+            .trim();
+          console.log('Checking category:', categoryName);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:44]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+          if (categories.includes(categoryName)) {
+            console.log('Found matching category:', categoryName);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:47]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+            const subMenu =
+              loadedCheerio(categoryElem).find('ul.sub-menu.m-sub');
+            console.log('Submenu items found:', subMenu.find('li').length);
+
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:51]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+            subMenu.find('li > a').each((_, novelElem) => {
+              const novelName = loadedCheerio(novelElem).text().trim();
+              const novelUrl = loadedCheerio(novelElem).attr('href');
+              console.log('Novel found:', novelName, novelUrl);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:55]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+                  path: novelUrl.replace(this.site, ''),
+                  cover: '',
+                });
+              }
+            });
+          }
+        },
+      );
+
+      console.log('Total novels found:', novels.length);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:69]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+      novels.forEach(novel => console.log('Novel:', novel.name, novel.path));
+
+      if (novels.length === 0) {
+        console.warn('No novels found, trying to parse everything...');
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:73]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+            const novelName = loadedCheerio(elem).text().trim();
+            novels.push({
+              name: novelName,
+              path: href.replace(this.site, ''),
+              cover: '',
+            });
+          }
+        });
+
+        console.log('Fallback total novels found:', novels.length);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:88]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+      }
+
+      if (showLatestNovels) {
+        novels.sort((a, b) => b.path.localeCompare(a.path));
+      }
+
+      return novels;
+    } catch (error) {
+      console.error('Error fetching popular novels:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:97]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+                path: chapterUrl.replace(this.site, ''),
+              });
+            }
+          });
+        });
+      }
+
+      if (chapters.length === 0) {
+        console.warn('No chapters found');
+      }
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:176]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+
+      return chapters.reverse();
+    } catch (error) {
+      console.error('Error parsing chapters:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:180]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+      throw error;
+    }
+  }
+
+  async getNovelCover(coverUrl: string): Promise<string> {
+    if (!coverUrl) return '';
+    try {
+      return await fetchFile(coverUrl);
+    } catch (error) {
+      console.error('Error fetching novel cover:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:190]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+    try {
+      const result = await fetchApi(url);
+      const body = await result.text();
+      const loadedCheerio = parseHTML(body);
+
+      const chapterTitle = loadedCheerio('h1.entry-title').text().trim();
+      const chapterContent = loadedCheerio('.post-body').html() || '';
+
+      if (!chapterContent) {
+        console.warn('No chapter content found');
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:211]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+      }
+
+      return `<h1>${chapterTitle}</h1>${chapterContent}`;
+    } catch (error) {
+      console.error('Error parsing chapter:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:216]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+            cover: novelCover,
+            // category: novelCategory,
+          });
+        }
+      });
+
+      if (novels.length === 0) {
+        console.warn('No search results found');
+      }
+
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:255]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 18]
```diff
@@ -0,0 +1,264 @@
...
+      return novels;
+    } catch (error) {
+      console.error('Error searching novels:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:258]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 19]
```diff
@@ -0,0 +1,264 @@
...
+            .text()
+            .match(/Author:\s*(.+)/i)?.[1]
+            ?.trim() || 'Unknown',
+        status: this.getNovelStatus(novelPath),
+        chapters: await this.parseChapters(loadedCheerio),
+      };
+
+      return novel;
+    } catch (error) {
+      console.error('Error parsing novel:', error);
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:127]:
```suggestion
```

File: src/plugins/english/dreambigtl.ts [hunk 20]
```diff
@@ -0,0 +1,235 @@
...
+      const result = await fetchApi(url);
+      const body = await result.text();
+      const loadedCheerio = parseHTML(body);
+
+      const novel: Plugin.SourceNovel = {
+        path: novelPath,
+        name: loadedCheerio('h1.entry-title').text().trim(),
+        cover: await this.getNovelCover(
+          loadedCheerio('.post-body img').first().attr('src') || '',
+        ),
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:98]: cover should be an url, not a base64 string

File: src/plugins/english/dreambigtl.ts [hunk 21]
```diff
@@ -0,0 +1,235 @@
...
+        status: this.getNovelStatus(novelPath),
+        chapters: await this.parseChapters(loadedCheerio),
+      };
+
+      return novel;
+    } catch (error) {
+      throw error;
+    }
+  }
+  async parseChapters(
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:114]:
```suggestion
  parseChapters(
```

File: src/plugins/english/dreambigtl.ts [hunk 22]
```diff
@@ -0,0 +1,235 @@
...
+            .text()
+            .match(/Author:\s*(.+)/i)?.[1]
+            ?.trim() || 'Unknown',
+        status: this.getNovelStatus(novelPath),
+        chapters: await this.parseChapters(loadedCheerio),
+      };
+
+      return novel;
+    } catch (error) {
+      throw error;
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:111]: If there nothing to do in catch, remove try catch block.

File: src/plugins/english/dreambigtl.ts [hunk 23]
```diff
@@ -0,0 +1,235 @@
...
+            }
+          });
+        });
+      }
+
+
+      return chapters.reverse();
+    } catch (error) {
+      throw error;
+    }
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:161]: If there nothing to do in catch, remove try catch block.

File: src/plugins/english/dreambigtl.ts [hunk 24]
```diff
@@ -0,0 +1,235 @@
...
+      const chapterTitle = loadedCheerio('h1.entry-title').text().trim();
+      const chapterContent = loadedCheerio('.post-body').html() || '';
+
+      if (!chapterContent) {
+      }
+
+      return `<h1>${chapterTitle}</h1>${chapterContent}`;
+    } catch (error) {
+      throw error;
+    }
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:194]: If there nothing to do in catch, remove try catch block.

File: src/plugins/english/dreambigtl.ts [hunk 25]
```diff
@@ -0,0 +1,235 @@
...
+            cover: novelCover,
+            // category: novelCategory,
+          });
+        }
+      });
+
+      return novels;
+    } catch (error) {
+      throw error;
+    }
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:231]: If there nothing to do in catch, remove try catch block.

File: src/plugins/english/dreambigtl.ts [hunk 26]
```diff
@@ -0,0 +1,235 @@
...
+    try {
+      const result = await fetchApi(url);
+      const body = await result.text();
+      const loadedCheerio = parseHTML(body);
+
+      const chapterTitle = loadedCheerio('h1.entry-title').text().trim();
+      const chapterContent = loadedCheerio('.post-body').html() || '';
+
+      if (!chapterContent) {
+      }
```
Comment (nyagami) [inline src/plugins/english/dreambigtl.ts:189]:
```suggestion
```

**Comment by r1di** [comment]
thanks @nyagami

**Pull request merged**